.github/workflows/build_gradle_sam.yml

```
@@ -0,0 +1,30 @@
+name: di-gradle-sam
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths:
+      - 'gradle-sam/*'
+      - '.github/workflows/build_gradle_sam.yml'
+  registry_package:
+    types:
+      - 'updated'
+
+  workflow_dispatch:
+    inputs:
+      image_tag:
+        description: "Tag used within docker repository to mark as unique"
+        required: false
+        type: string
+
+jobs:
+  build-and-push:
+    if: github.event.registry_package.package.name == 'aws-sam'
```

This conditional will stop us building on changes to the gradle-sam dockerfile.
```suggestion
    if: github.event.type != package || github.event.registry_package.package.name == 'aws-sam'
```
not terribly readable tho